olp-cpp-sdk-core/tests/cache/DefaultCacheTest.cpp

```diff
@@ -369,4 +369,266 @@ TEST(DefaultCacheTest, ValueGreaterThanMemCacheLimit) {
+    ASSERT_FALSE(key1DataRead);
+  }
+  {
+    SCOPED_TRACE("RemoveWithPrefix.");
+    olp::cache::DefaultCache cache(settings);
+    ASSERT_EQ(olp::cache::DefaultCache::Success, cache.Open());
+    ASSERT_TRUE(cache.Clear());
+    std::vector<unsigned char> binary_data = {1, 2, 3};
+    std::string key1DataString{"this is key1's data"};
+    cache.Put(
+        "key1", key1DataString, [=]() { return key1DataString; },
+        (std::numeric_limits<time_t>::max)());
+    cache.Put("somekey1",
+              std::make_shared<std::vector<unsigned char>>(binary_data),
+              (std::numeric_limits<time_t>::max)());
+    cache.Put("somekey2",
+              std::make_shared<std::vector<unsigned char>>(binary_data),
+              (std::numeric_limits<time_t>::max)());
+
+    ASSERT_FALSE(cache.Get("key1", [](const std::string& data) { return data; })
+                     .empty());
+    ASSERT_TRUE(cache.Get("somekey1"));
+    ASSERT_TRUE(cache.Get("somekey2"));
+
+    ASSERT_TRUE(cache.RemoveKeysWithPrefix("invalid_prefix"));
+    ASSERT_FALSE(cache.Get("key1", [](const std::string& data) { return data; })
+                     .empty());
+    ASSERT_TRUE(cache.Get("somekey1"));
+    ASSERT_TRUE(cache.Get("somekey2"));
+
+    ASSERT_TRUE(cache.RemoveKeysWithPrefix("key"));
+    ASSERT_TRUE(cache.Get("key1", [](const std::string& data) { return data; })
+                    .empty());
+    ASSERT_TRUE(cache.Get("somekey1"));
+    ASSERT_TRUE(cache.Get("somekey2"));
+
+    ASSERT_TRUE(cache.RemoveKeysWithPrefix("somekey"));
+    ASSERT_FALSE(cache.Get("somekey1"));
+    ASSERT_FALSE(cache.Get("somekey2"));
+  }
+  {
+    SCOPED_TRACE("Clear.");
+    olp::cache::DefaultCache cache(settings);
+    ASSERT_EQ(olp::cache::DefaultCache::Success, cache.Open());
+    ASSERT_TRUE(cache.Clear());
+    std::vector<unsigned char> binary_data = {1, 2, 3};
+
+    bool put_result = cache.Put(
+        "key1", std::make_shared<std::vector<unsigned char>>(binary_data),
+        (std::numeric_limits<time_t>::max)());
+    ASSERT_TRUE(put_result);
+
+    ASSERT_TRUE(cache.Clear());
+    auto key1DataRead = cache.Get("key1");
+    ASSERT_FALSE(key1DataRead);
+  }
+  {
+    SCOPED_TRACE("Load disk cache.");
+    olp::cache::DefaultCache cache(settings);
+    ASSERT_EQ(olp::cache::DefaultCache::Success, cache.Open());
+    ASSERT_TRUE(cache.Clear());
+    std::vector<unsigned char> binary_data = {1, 2, 3};
+    bool put_result = cache.Put(
+        "key1", std::make_shared<std::vector<unsigned char>>(binary_data),
+        (std::numeric_limits<time_t>::max)());
+    ASSERT_TRUE(put_result);
+    auto key1DataRead = cache.Get("key1");
+    ASSERT_TRUE(key1DataRead);
+
+    cache.Close();
+
+    ASSERT_EQ(olp::cache::DefaultCache::Success, cache.Open());
+    key1DataRead = cache.Get("key1");
+    ASSERT_TRUE(key1DataRead);
+    ASSERT_EQ(*key1DataRead, binary_data);
+  }
+}
+
+TEST(DefaultCacheTest, EvictionPolicykLeastRecentlyUsed) {
```

Comment:
What exactly are you testing here? I don't see any difference?
I don't see any real unit test here. Unit test should test that the LRU cache is created on open, that it contains all key, that adding new key to the cache also adds it to the LRU, that removing one key from cache also adds it to the LRU. Please implement such a test.

Reply:
Eviction policy is different:-) I have moved tests into separate function.
Existing tests are not covering well mutable cache use, so the idea was to make  couple of tests for each DefaultCache method and check them with different eviction policies.
I can't check is LRU cache actually created and used through DefaultCache interface, only validate nothing is broken with different options.
I think, it can be tested only by eviction strategy tests, but this is out of scope of the PR.

Reply:
Unit tests is not an API test. It is a class test. So if you need to access the internals of the DefaultCache in unit tests, then you should do it. Prepare the DefaultCache class so that a derived class can access needed internals, e.g. trough protected section helpers.

Reply:
Added some tests to check lru cache directly.